openfisca_core/scripts/migrations/v16_2_to_v17/xml_to_yaml.py

```diff
@@ -80,7 +80,13 @@ def transform_values_history(children, value_format):
         elif child.tag == 'END':
             values[date] = {'value': None}
         elif child.tag == 'PLACEHOLDER':
-            values[date] = "expected"
+            values[date] = 'expected'
+
+        if 'reference' in child.attrib:
+            if values[date] == 'expected':
+                values[date] = {'expected': None, 'reference': child.attrib['reference']}
+            else:
+                values[date]['reference'] = child.attrib['reference']
```

Comment:
Why is this not applied to all cases ? What is the different between this one and the one above ?

Reply:
If tag is END or VALUE, values[date] is a dict. If tag is PLACEHOLDER, it is the string "expected".
This object is then serialized in YAML. The string "expected" is prefered over {"expected": None} because the following YAML is ugly : 
```yaml
    2017-01-01:
        expected: None
```